bee-protocol/src/workers/metrics.rs

```diff
@@ -61,6 +106,28 @@ impl<N: Node> Worker<N> for MetricsWorker {
             info!("Stopped.");
         });
 
+        node.spawn::<Self, _, _>(|shutdown| async move {
+            let mut ticker =
+                ShutdownStream::new(shutdown, IntervalStream::new(interval(MEMORY_METRIC_UPDATE_INTERVAL)));
+
+            while ticker.next().await.is_some() {
+                process_metrics.update().await;
+            }
+        });
+
+        node.spawn::<Self, _, _>(|shutdown| async move {
+            info!("Running prometheus client.");
+
+            let serve = bee_metrics::serve_metrics(config.bind_address, registry);
```

Comment:
To be honest I don't think these should be here. They have nothing to do with the protocol. Can't we have a worker directly in the bee-metrics crate ?